src/Test/Utilities/Portable/Platform/CoreClr/TestExecutionLoadContext.cs

```diff
@@ -21,73 +21,33 @@ namespace Roslyn.Test.Utilities.CoreClr
 {
     internal sealed class TestExecutionLoadContext : AssemblyLoadContext
     {
-        private readonly static ImmutableDictionary<string, string> s_platformAssemblyPaths = GetPlatformAssemblyPaths();
-        private readonly static Dictionary<string, Assembly> s_loadedPlatformAssemblies = new Dictionary<string, Assembly>(StringComparer.Ordinal);
-
-        private readonly Dictionary<string, ModuleData> _dependencies;
+        private readonly ImmutableDictionary<string, ModuleData> _dependencies;
 
         public TestExecutionLoadContext(IList<ModuleData> dependencies)
         {
-            _dependencies = new Dictionary<string, ModuleData>(dependencies.Count, StringComparer.Ordinal);
-            foreach (var dep in dependencies)
-            {
-                _dependencies.Add(dep.FullName, dep);
-            }
+            _dependencies = dependencies
+                .Select(x => new KeyValuePair<string, ModuleData>(x.FullName, x))
+                .ToImmutableDictionary(StringComparer.Ordinal);
```

Comment:
Could this use the following instead?
```C#
dependencies.ToImmutableDictionary(
    d => d.FullName,
    StringComparer.Ordinal);
```

Reply:
Sure thing. I have a follow up PR in this area. Will make the change there.